nplab/instrument/message_bus_instrument.py

```diff
@@ -111,12 +111,11 @@ def query(self,queryString,multiline=False,termination_line=None,timeout=None):
             self.flush_input_buffer()
             self.write(queryString)
             if self.ignore_echo == True: # Needs Implementing for a multiline read!
-                first_line = self.readline(timeout).strip()
-                if first_line == queryString:
-                    return self.readline(timeout).strip()
-                else:
-                    print 'This command did not echo!!!'
-                    return first_line
+                    first_line = self.readline(timeout).strip()
+                    if first_line == queryString:
+                        self._logger.warn('%s did not echo' % queryString)
+                        return self.readline(timeout).strip()
+ print 'This command did not echo!!!'
     
```

Comment:
```suggestion
```suggestion
        return first_line
```
```